sips/sip-x/sip-x-improved-stacking.md

```diff
@@ -0,0 +1,205 @@
+- locking Stacks tokens is protected by a new type of post conditions, enabling stacking through a contract in a single transaction.
+
+# Introduction
+
+## Current Situation
+
+Currently, the stacking protocol has a few aspects that make using and integration stacking harder than it could be:
+
+- There are two groups of stackers: solo stackers and delegated stackers. They use different sets of PoX contract functions.
+
+- When using a contract for stacking, the contract needs to be added as an allowed pox contract. This requires a separate transaction.
+
+- Solo stackers do all transactions with their cold wallet holding the whole STX balance. They need to make a transaction at least once every 12 cycles (6 months). In contrast, delegated stackers need to do only a single transaction with their wallet for the entire stacking.
+
+- Signers make off-chain agreements with pool operators regarding revenue sharing.
+
+## Glossary
+
+| Term                 | Definition                                                                                                                                               |
+| -------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Cooldown Cycle**   | The period of 1 stacking cycle where stackers cannot stack their STX tokens because their tokens are unlocked only at the beginning of a stacking cycle. |
+| **Signer**           | The node operators that verify and confirm proposed blocks by miners.                                                                                    |
+| **Delegation (old)** | delegating the management of stacking.                                                                                                                   |
+| **Delegation (new)** | delegating the voting power for proposed blocks.                                                                                                         |
+| **PoX Contract**     | The smart contract that users interact with to lock their tokens.                                                                                        |
+
+## Problem Statement
+
+The current stacking process has two different paths, one for solo stackers, one for delegated stackers,
+resulting in a more complex user experience and more complex code.
+
+The process is defined by a prepare phase of 100 bitcoin blocks that is used to find an anchor block
+for the next stacking cycle. Stackers must lock
+their Stacks tokens before the
+prepare phase, i.e. before the 2000th block of the current stacking cycle. Furthermore, Stacks
+tokens are locked for a locking period that always ends at the beginning of a cycle, i.e. after the
+prepare phase. Therefore, the current implementation of Stacking includes a period where Stacks
+tokens are unlocked and cannot earn stacking rewards (Cooldown Cycle).
```

Comment:
...includes a period during which Stacks tokens are unlocked and ineligible for stacking rewards (Cooldown Cycle).

(suggestion to improve clarity)

Reply:
:+1: